go-controller/pkg/ovn/controller/services/node_tracker.go

```diff
@@ -147,18 +161,18 @@ func (nt *nodeTracker) updateNodeInfo(nodeName, switchName, routerName string, n
 	}
 
 	nt.nodes[nodeName] = ni
-	nt.Unlock()
 
-	klog.Infof("Node %s switch + router changed, syncing services", nodeName)
 	// Resync all services
-	nt.resyncFn()
+	nt.resyncFn(nt.allNodes())
 }
 
 // removeNodeWithServiceReSync removes a node from the LB -> node mapper
 // *and* forces full reconciliation of services.
 func (nt *nodeTracker) removeNodeWithServiceReSync(nodeName string) {
 	nt.removeNode(nodeName)
-	nt.resyncFn()
+	nt.Lock()
+	nt.resyncFn(nt.allNodes())
+	nt.Unlock()
```

Comment:
nit: I would have rather opted to deep copy in `allNodes` rather than locking during `resyncFn`. But not a huge thing, it's all in the same package and we are aware that what's going on in `resyncFn` should not block the node tracker for too long.